File: ALDS/2_B.js
'use strict';

function Main(input) {
	input = input.split('\n');
	const N = Number(input[0]);
	const ary = input[1].split(' ').map(Number);

	let n = 0;
	function selectionSort(A, N) {
		for (let i = 0; i < N - 1; i++) {
			let minj = i;
			for (let j = i; j <= N - 1; j++) {
				if (A[j] < A[minj]) {
					minj = j;
				}
			}
			if (A[i] !== A[minj]) n++;
			[A[i], A[minj]] = [A[minj], A[i]];
		}
		return A;
	}
	const result = selectionSort(ary, N);
	console.log(result.join(' '));
	console.log(n);
}

// Main(require('fs').readFileSync('/dev/stdin', 'utf8'));
Main(require('fs').readFileSync('./ALDS/test.txt', 'utf8'));

File: ALDS1/2_B.py
N = int(input())
A = list(map(int, input().split()))

c = 0
for i in range(N):
    minj = i
    for j in range(i, N):
        if A[j] < A[minj]:
            minj = j
    if minj != i:
        A[i], A[minj] = A[minj], A[i]
        c += 1

print(" ".join([str(a) for a in A]))
print(c)
